### PR TITLE
Add backend cleanup feature

### PR DESCRIPTION
**Pull request by anisurrahman75**
Signed-off-by: anisurrahman75 <sunny.cse7575@gmail.com>

### CODE REVIEW

File: apis/constant.go [hunk 1]
```diff
@@ -68,10 +68,11 @@ const (
...
 
-	KubeStashBackupJobClusterRole       = "kubestash-backup-job"
-	KubeStashRestoreJobClusterRole      = "kubestash-restore-job"
-	KubeStashCronJobClusterRole         = "kubestash-cron-job"
-	KubeStashLocalBackendJobClusterRole = "kubestash-localBackend-job"
+	KubeStashBackupJobClusterRole         = "kubestash-backup-job"
+	KubeStashRestoreJobClusterRole        = "kubestash-restore-job"
+	KubeStashCronJobClusterRole           = "kubestash-cron-job"
+	KubeStashLocalBackendJobClusterRole   = "kubestash-localBackend-job"
+	KubeStashBackendCleanupJobClusterRole = "kubestash-BackendCleanup-job"
```
Comment (hmsayem) [inline apis/constant.go:75]: `KubeStashLocalBackendJobClusterRole   = "kubestash-local-backend-job"`
`KubeStashBackendCleanerJobClusterRole = "kubestash-backend-cleaner-job"`

File: apis/storage/v1alpha1/backupstorage_types.go [hunk 2]
```diff
@@ -100,6 +100,10 @@ type BackupStorageStatus struct {
 	// Conditions represents list of conditions regarding this BackupStorage
 	// +optional
 	Conditions []kmapi.Condition `json:"conditions,omitempty"`
+
+	// Job holds the cleanup jobs Information that's created regarding this BackupStorage
```
Comment (hmsayem) [inline apis/storage/v1alpha1/backupstorage_types.go:104]: // Job holds the cleaner jobs information for this BackupStorage

File: apis/storage/v1alpha1/repository_types.go [hunk 3]
```diff
@@ -115,6 +115,10 @@ type RepositoryStatus struct {
 	// Conditions represents list of conditions regarding this Repository
 	// +optional
 	Conditions []kmapi.Condition `json:"conditions,omitempty"`
+
+	// Job holds the cleanup jobs Information that's created regarding this Repository
```
Comment (hmsayem) [inline apis/storage/v1alpha1/repository_types.go:119]: // Job holds the cleaner jobs information for this repository

File: apis/storage/v1alpha1/snapshot_helpers.go [hunk 4]
```diff
@@ -34,6 +34,11 @@ func (_ Snapshot) CustomResourceDefinition() *apiextensions.CustomResourceDefini
 }
 
 func (s *Snapshot) CalculatePhase() SnapshotPhase {
```
Comment (hmsayem) [inline apis/storage/v1alpha1/snapshot_helpers.go:36]:
```go
func (s *Snapshot) CalculatePhase() SnapshotPhase {
	if kmapi.IsConditionFalse(s.Status.Conditions, TypeSnapshotMetadataUploaded) ||
		kmapi.IsConditionFalse(s.Status.Conditions, TypeRecentSnapshotListUpdated) {
		return SnapshotFailed
	}
	
	if s.GetComponentsPhase() == SnapshotPending {
		return SnapshotPending
	}
	
	if kmapi.HasCondition(s.Status.Conditions, TypeSnapshotMetadataUploaded) {
		return s.GetComponentsPhase()
	}
	
	return  SnapshotRunning
}
```

**Review by hmsayem — CHANGES_REQUESTED** [review]

File: apis/storage/v1alpha1/backupstorage_types.go [hunk 5]
```diff
@@ -100,6 +100,10 @@ type BackupStorageStatus struct {
 	// Conditions represents list of conditions regarding this BackupStorage
 	// +optional
 	Conditions []kmapi.Condition `json:"conditions,omitempty"`
+
+	// Job holds the cleaner jobs information for this BackupStorage
+	// +optional
+	Job string `json:"job,omitempty"`
```
Comment (hmsayem) [inline apis/storage/v1alpha1/backupstorage_types.go:106]: `Job` -> `cleanerJob`

File: apis/storage/v1alpha1/repository_types.go [hunk 6]
```diff
@@ -115,6 +115,10 @@ type RepositoryStatus struct {
 	// Conditions represents list of conditions regarding this Repository
 	// +optional
 	Conditions []kmapi.Condition `json:"conditions,omitempty"`
+
+	// Job holds the cleaner jobs information for this repository
+	// +optional
+	Job string `json:"job,omitempty"`
```
Comment (hmsayem) [inline apis/storage/v1alpha1/repository_types.go:121]: `Job` -> `cleanerJob`

File: apis/variables.go [hunk 7]
```diff
@@ -46,6 +46,7 @@ const (
 	TaskName = "taskName"
 
 	Namespace      = "namespace"
+	Name           = "name"
```
Comment (hmsayem) [inline apis/variables.go:49]: remove this variable

**Review by hmsayem — CHANGES_REQUESTED** [review]

File: apis/storage/v1alpha1/snapshot_helpers.go [hunk 8]
```diff
@@ -162,6 +169,7 @@ func GenerateSnapshotName(repoName, backupSession string) string {
 
 func (s *Snapshot) OffshootLabels() map[string]string {
 	newLabels := make(map[string]string)
+	newLabels[apis.KubeStashInvokerKind] = ResourceKindBackupStorage
```
Comment (hmsayem) [inline apis/storage/v1alpha1/snapshot_helpers.go:172]: `ResourceKindSnapshot`

File: apis/storage/v1alpha1/repository_types.go [hunk 9]
```diff
@@ -115,6 +115,10 @@ type RepositoryStatus struct {
 	// Conditions represents list of conditions regarding this Repository
 	// +optional
 	Conditions []kmapi.Condition `json:"conditions,omitempty"`
+
+	// CleanerJob holds the cleaner jobs information for this Repository
+	// +optional
+	CleanerJob string `json:"cleanerJob,omitempty"`
```
Comment (hmsayem) [inline apis/storage/v1alpha1/repository_types.go:121]: add `invokerKind` label  repoisotry

**Review by hmsayem — CHANGES_REQUESTED** [review]

**Review by hmsayem — APPROVED** [review]

**Review by ishtiaqhimel — APPROVED** [review]

**Pull request merged**